File: app-frontend/lib/budget.ts
export function calculateBudget(
  income: number,
  fixed: number,
  variable: number
) {
  const totalExpenses = fixed + variable;
  const savings = Math.max(income - totalExpenses, 0);

  return {
    needs: fixed,
    wants: variable,
    savings,
    savingsRate: income > 0 ? (savings / income) * 100 : 0,
  };
}

File: model-backend/services/budget.py
def calculate_budget(income, fixed, variable):
    total_expenses = fixed + variable
    savings = max(income - total_expenses, 0)

    return {
        "needs": fixed,
        "wants": variable,
        "savings": savings,
        "savings_rate": round((savings / income) * 100, 2) if income else 0
    }
